.github/workflows/publish-rock.yaml

```
@@ -7,34 +7,32 @@ on:
         description: "Name of the artifact from which the ROCK will be downloaded."
         required: true
         type: string
-      rock-reference:
-        description: "Rock reference to be used in order to copy the image to docker cache.
-          It consists of <rock-name>:<rock-version>."
-        required: true
-        type: string
       rock-filename:
         description: "Filename of the .rock file"
         required: true
         type: string
-      registry:
-        description: "Registry where the image will be published"
+      full-image-tag:
+        description: "Full image tag to publish (e.g. docker.io/org/rock:version-commit)"
         required: true
         type: string
       source-branch:
-        description: Github branch to checkout. If blank, it will use default values.
-        default: '' # For default behaviour, see https://github.com/actions/checkout/tree/v4/#usage
+        description: "GitHub branch to checkout. If blank, it will use default values."
+        default: ''
         required: false
         type: string
+
 jobs:
   publish-rock:
-    name: Publish ${{ inputs.rock-reference }}
+    name: Publish ${{ inputs.full-image-tag }}
     runs-on: ubuntu-22.04
+    outputs:
+      published-image: ${{ steps.push.outputs.image }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
           ref: ${{ inputs.source-branch }}
-      
+
       - name: Install Rockcraft
         run: sudo snap install rockcraft --classic --edge
```

nitpick: I dont think we need to install `--edge` at this point
```suggestion
        run: sudo snap install rockcraft --classic
```